src/Concordium/Client/Runner.hs

```diff
@@ -1269,6 +1295,61 @@ getCryptographicParametersOrDie bhInput = do
         StatusInvalid -> logFatal ["GRPC response contained an invalid status code."]
         RequestFailed err -> logFatal ["I/O error: " <> err]
 
+-- | Compute the time of the first payday after a given time.
+--  This is used for determining the time at which a cooldown will actually elapse.
+firstPaydayAfter ::
+    -- | Time of the next payday.
+    UTCTime ->
+    -- | Duration of an epoch
+    Types.Duration ->
+    -- | Length of a payday.
+    Types.RewardPeriodLength ->
+    -- | Time at which the cooldown expires.
+    UTCTime ->
+    UTCTime
+firstPaydayAfter nextPayday epochDuration (Types.RewardPeriodLength ep) cooldownEnd =
+    if cooldownEnd <= nextPayday
+        then nextPayday
+        else
+            let timeDiff = Clock.diffUTCTime cooldownEnd nextPayday
+                paydayLength = Types.durationToNominalDiffTime (fromIntegral ep * epochDuration)
+                mult :: Word = ceiling (timeDiff / paydayLength)
+            in  Clock.addUTCTime (fromIntegral mult * paydayLength) nextPayday
```

Comment:
I'm not sure I fully understand this calculation. `paydayLength` sounds like it should be equal to the `ep` input variable which is described as "Length of a payday" but it seems like it is not equal. Could you maybe add a comment explaining why the calculation is done like this? Why is this the correct way to calculate it?